src/app/shared/services/firebase.service.ts

```diff
@@ -1,9 +1,231 @@
 import { Injectable } from '@angular/core';
+import {
+  AngularFirestore,
+  AngularFirestoreCollection,
+  AngularFirestoreDocument,
+  DocumentReference,
+  QueryFn
+} from "@angular/fire/firestore";
+import {User} from "../models/user.model";
+import {Patient} from "../models/patient.model";
+import {Observable} from "rxjs";
+import actions from "@angular/fire/schematics/deploy/actions";
+import {map} from "rxjs/operators";
+import {Medicine} from "../models/medicine.model";
+import {Courier} from "../models/courier.model";
+import {Order} from "../models/order.model";
 
 @Injectable({
   providedIn: 'root'
 })
 export class FirebaseService {
 
-  constructor() { }
+  users: User[];
+
+  constructor(private firestore: AngularFirestore) {
+
+    //FIXME Delete examples once we start to use the service
+
+    let patient = new Patient();
+    patient.name = "name";
+    patient.idNumber = "123456";
+    patient.birth_date = new Date();
+    patient.email = "email@email.com"
+    patient.address = "Calle Falsa 123";
+    patient.phoneNumber = "=573213331112"
+
+    /*this.createPatient(patient)
+        .then(value => console.debug("Created"))
+        .catch(reason => console.error("[error]"));*/
+
+    this.getPatients( ref => ref.where("idNumber", "==", "123456"))
+        .subscribe(patients => {
+          // from another service we would use:
+          // this.patients = value;
+          console.debug("[DEBUG] WITH query function", patients);
+        });
+
+    this.getPatients()
+        .subscribe(patients => {
+          // from another service, with patients: Observable<Patient>, we would use:
+          // this.patients = value;
+          console.debug("[DEBUG] WITHOUT query function", patients);
+        });
+
+
+    patient.id = "vvmpDJp1aRs0pIS0sSKD"; // ID comes when we use getPatients
+    patient.phoneNumber = "+573213213211"
+    this.updatePatient(patient)
+        .then(value => {
+          console.debug("[DEBUG] Updated", value);
+        })
+        .catch(reason => {
+          console.debug("[DEBUG] Error updating", reason);
+        });
+
+    this.deletePatient("tI4e9H6Jkr9U5AdIX0Kp") // A valid id
+        .then(value => {
+          console.debug("[DEBUG] Deleted", value);
+        })
+        .catch(reason => {
+          console.debug("[DEBUG] Error deleting", reason);
+        });
+
+  }
+
+  getUsers(){
+    this.firestore.collection<User>('users').valueChanges().subscribe((data)=>{
+      this.users = data;
+      console.debug(this.users);
+    });
```

Comment:
Fort curiosity, why need to keep listening to the changes for Users

Reply:
It's the way Firebase is used